### ISSUE TITLE
[cli]: External database extensions fail to load in released CLI

### ISSUE THREAD

**Issue by prrao87** [kuzudb/kuzu#3478]
Kùzu version: 0.4.1
Extension version: 0.3.1

Both the duckdb and postgres extensions fail to load successfully via the CLI on macOS. The issue isn't present when the extensions are loaded via the Python client.

```
kuzu> INSTALL duckdb;
-----------------------------------------
| result                                |
-----------------------------------------
| Extension: duckdb has been installed. |
-----------------------------------------
(1 tuple)
(1 column)
Time: 0.07ms (compiling), 884.58ms (executing)
kuzu> LOAD EXTENSION duckdb;
Error: IO exception: Extension "/Users/prrao/.kuzu/extension/0.3.1/osx_arm64/libduckdb.kuzu_extension" could not be loaded.
Error: dlopen(/Users/prrao/.kuzu/extension/0.3.1/osx_arm64/libduckdb.kuzu_extension, 0x0006): symbol not found in flat namespace '__ZN4kuzu4main13ClientContext16getMemoryManagerEv'
```

```
kuzu> INSTALL postgres;
-------------------------------------------
| result                                  |
-------------------------------------------
| Extension: postgres has been installed. |
-------------------------------------------
(1 tuple)
(1 column)
Time: 0.04ms (compiling), 720.29ms (executing)
kuzu> LOAD EXTENSION postgres;
Error: IO exception: Extension "/Users/prrao/.kuzu/extension/0.3.1/osx_arm64/libpostgres.kuzu_extension" could not be loaded.
Error: dlopen(/Users/prrao/.kuzu/extension/0.3.1/osx_arm64/libpostgres.kuzu_extension, 0x0006): symbol not found in flat namespace '__ZN4kuzu4main13ClientContext16getMemoryManagerEv'
```

**Comment by mewim** [comment]
I have published v0.4.2 with the fix and submitted a PR to Homebrew repo: https://github.com/Homebrew/homebrew-core/pull/171711. This can be closed after the PR is merged.

**Comment by mewim** [comment]
<img width="1394" alt="Screenshot 2024-05-14 at 5 48 10 PM" src="https://github.com/kuzudb/kuzu/assets/14037726/00491285-9450-40c6-bbc6-c1965ed780fe">
Can confirm that kuzu v0.4.2 from Homebrew can now load extensions without the error.
